src/system-core.js

```diff
@@ -199,18 +197,23 @@ export function getOrCreateLoad (loader, id, firstParentUrl) {
     // On execution, L, I, E cleared
 
     // Promise for top-level completion
-    C: undefined
+    C: undefined,
+
+    // parent instantiator / executor
+    p: undefined
   };
 }
 
-function instantiateAll (loader, load, loaded) {
+function instantiateAll (loader, load, parent, loaded) {
   if (!loaded[load.id]) {
     loaded[load.id] = true;
     // load.L may be undefined for already-instantiated
     return Promise.resolve(load.L)
     .then(function () {
+      if (load.e !== null)
+        load.p = parent;
```

Comment:
Just thinking about this a little more carefully, this should probably be:

```suggestion
      if (load.e !== null && (!load.p || load.p.e === null))
        load.p = parent;
```